src/formatter-settings/index.ts

```diff
@@ -4,27 +4,347 @@
+            content: content,
+        });
+    }
+
+    private async changeProfileSetting(id: string, value: string, document: vscode.TextDocument): Promise<void> {
+        const profileElement = this.profileElements.get(id);
+        if (!profileElement) {
+            // add a new setting not exist in the profile
+            if (!this.profileSettingsArray.length) {
+                return;
+            }
+            const cloneElement = this.profileSettingsArray[this.profileSettingsArray.length - 1].cloneNode();
+            const originString: string = new XMLSerializer().serializeToString(cloneElement);
+            cloneElement.setAttribute("id", id);
+            cloneElement.setAttribute("value", value);
+            const newString: string = new XMLSerializer().serializeToString(cloneElement);
+            const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+            edit.insert(document.uri, new vscode.Position(cloneElement.lineNumber - 1, cloneElement.columnNumber - 1 + originString.length), EOL + " ".repeat(cloneElement.columnNumber - 1) + newString);
+            vscode.workspace.applyEdit(edit);
+            cloneElement.lineNumber++;
+            this.profileElements.set(id, cloneElement);
+            this.profileSettingsArray.push(cloneElement);
+        } else {
+            // edit a current setting in the profile
+            const originSetting: string = new XMLSerializer().serializeToString(profileElement);
+            const line: number = profileElement.lineNumber;
+            const column: number = profileElement.columnNumber;
+            await profileElement.setAttribute("value", value);
+            const newSetting: string = new XMLSerializer().serializeToString(profileElement);
+            const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+            edit.replace(document.uri, new vscode.Range(new vscode.Position(line - 1, column - 1), new vscode.Position(line - 1, column - 1 + originSetting.length)), newSetting);
+            vscode.workspace.applyEdit(edit);
+        }
+    }
+
+    public async showFormatterSettingsEditor() {
+        if (!this.settingsUrl) {
+            vscode.window.showInformationMessage("No available Java Formatter Profile found, do you want to create a default profile?",
+                "Yes", "No").then((messageResult) => {
+                    if (messageResult === "Yes") {
+                        this.addFormatter();
+                    }
+                });
+        } else {
+            if (this.isRemote(this.settingsUrl)) {
+                await vscode.window.showInformationMessage("The active formatter profile is in remote URL, please change to a local file and try to reopen this setting again.");
+                return;
+            }
+            const profilePath = await this.getProfilePath(this.settingsUrl);
+            vscode.commands.executeCommand("vscode.openWith", vscode.Uri.file(profilePath), "java.formatterSettingsEditor");
```

Comment:
What will happen if the profilePath does not exist?

Reply:
It will throw an unexpected error now. Will check if the profile exists.